lalrpop/src/build/mod.rs

```diff
@@ -157,6 +165,14 @@ fn needs_rebuild(lalrpop_file: &Path, rs_file: &Path) -> io::Result<bool> {
                                   -> bool {
         true
     }
+
+    fn compare_lalrpop_version(rs_file: &Path) -> io::Result<bool> {
+        let mut input_str = String::new();
+        let mut f = io::BufReader::new(try!(fs::File::open(&rs_file)));
+        try!(f.read_line(&mut input_str));
+
+        Ok(input_str.trim() != LALRPOP_VERSION_HEADER)
```

Comment:
Very nice.

I wonder if there is some way we can test this... well, we're not testing it now anyway I guess. We should think about it for the future though!